### PR TITLE
cmake: Use FetchContent instead of submodules

### PR DESCRIPTION
**Pull request by bwrsandman**
Follows #116

cmake: Use FetchContent instead of submodules

Update minimum cmake to 3.14
Remove all submodules.
Fetch content fetches dependencies at cmake config time and puts them in
the cmake build directory.

### CODE REVIEW

**Comment by bwrsandman** [comment]
Open discussion:
Do we want to consider using FetchContent instead of Hunter for the all or some of the following deps:
* SDL2
* spdlog
* glm
* EnTT (which is both gotten on both?)

**Comment by bwrsandman** [comment]
bionic: You are running version 3.10.2
disco: You are running version 3.13.4

FetchContent can run on 3.11 but has better features at 3.14

**Comment by bwrsandman** [comment]
Here's a proposed update to the docker image
https://github.com/openblack/openblack-docker/pull/2

File: ci/Dockerfile.ubuntu.template [hunk 1]
```diff
@@ -13,6 +13,12 @@ RUN apt-get update && \
   libspdlog-dev \
   libwayland-dev
 
+# Install cmake 3.15
+RUN wget https://github.com/Kitware/CMake/releases/download/v3.15.3/cmake-3.15.3-Linux-x86_64.sh && \
+  chmod +x cmake-3.15.3-Linux-x86_64.sh && \
+  apt remove cmake && \
+  ./cmake-3.15.3-Linux-x86_64.sh --skip-license --prefix=/usr/local
+
```
Comment (bwrsandman) [inline ci/Dockerfile.ubuntu.template]: This file is not used?

**Comment by bwrsandman** [comment]
Job | Status
-- | --
310.1 bionic: gcc | failing due to cmake version being less than 3.11 https://github.com/openblack/openblack-docker/pull/2 and not having git https://github.com/openblack/openblack-docker/pull/3
310.3 disco: clang | failing due to not having git https://github.com/openblack/openblack-docker/pull/3
310.4 fedora rawhide | passing
310.5 fedora 31: mingw-w64 | passing

**Comment by bwrsandman** [comment]
I understand it LTS support is more important and it won't really affect my other PRs

**Comment by bwrsandman** [comment]
I backported FetchContent to `cmake/future` this should work with ubuntu LTS.
I also fixed a bug in the ci system which was using the `-B` command which is not available on LTS.

**Comment by bwrsandman** [comment]
All ubuntu now have the same error:
```
error: could not find git for clone of entt-populate
```

Job | Status
-- | --
310.1 bionic: gcc | failing due to not having git https://github.com/openblack/openblack-docker/pull/3
310.2 disco: gcc | failing due to not having git https://github.com/openblack/openblack-docker/pull/3
310.3 disco: clang | failing due to not having git https://github.com/openblack/openblack-docker/pull/3
310.4 fedora rawhide | passing
310.5 fedora 31: mingw-w64 | passing

**Comment by bwrsandman** [comment]
@NeroBurner I tested in the lts docker and it was building in-source

**Comment by NeroBurner** [comment]
do you have the command you tested with? You have to use `cmake -H. -B_build` OR `cmake -S . -B _build`. Notice the space between the arguments

**Comment by bwrsandman** [comment]
@NeroBurner It was few days ago, but I was going with the commands in the .yaml file...

**Pull request merged**